### PR TITLE
fix #494 Use nix configuration in docker build

### LINKED ISSUE #494: Proposal: remove docker script from the main repo
We don't test the docker script in https://github.com/scala-native/scala-native/tree/master/docker which means it's likely outdated and/or broken already. Given that we've had relatively straightforward setup on linux for a while now, I'd propose to remove it from the main repo.

### PR DESCRIPTION
**Pull request by MasseGuillaume**
depends on #501

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,6 +13,24 @@ lazy val baseSettings = Seq(
   sources in doc in Compile := Nil // doc generation currently broken
 )
 
+addCommandAlias(
+  "fullRebuild",
+  Seq(
```
Comment (densh) [inline build.sbt]: I'd add "clean" as the first line.

File: build.sbt [hunk 2]
```diff
@@ -13,6 +13,24 @@ lazy val baseSettings = Seq(
...
+    "nscplugin/publishLocal",
+    "nativelib/publishLocal",
+    "publishLocal",
+    "sandbox/run",
+    "demoNative/run",
+    "tests/run",
+    "tools/test",
+    "benchmarks/run",
+    "scripted",
+    "publishSnapshot"
```
Comment (densh) [inline build.sbt]: Tests invocations and publishing should probably go separately as rebuild command would be useful for local development too.

File: build.sbt [hunk 3]
```diff
@@ -13,6 +13,24 @@ lazy val baseSettings = Seq(
   sources in doc in Compile := Nil // doc generation currently broken
 )
 
+addCommandAlias(
+  "fullRebuild",
```
Comment (densh) [inline build.sbt]: "rebuild"

File: unit-tests/src/main/scala/java/lang/ExceptionSuite.scala [hunk 4]
```diff
@@ -7,29 +7,20 @@ object ExceptionSuite extends tests.Suite {
...
-    ).mkString("\n")
-    // It's startsWith and not == as there could be additional stack
-    // activations before main in the stack trace. For example on linux
-    // there would be additional __libc_start_main and _start frames.
-    assert(sw.toString.startsWith(expected))
+    )
+
+    val stackTrace = sw.toString
+
+    assert(segments.forall(stackTrace.contains))
```
Comment (densh) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: This doesn't look that good to me. What's the full text of the stack trace that you're getting?

Comment (densh) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: @MasseGuillaume ?

Comment (MasseGuillaume) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]:
```
* java.lang.ExceptionSuite$
java.lang.Exception
        at java.lang.ExceptionSuite$$anonfun$1::apply$mcV$sp_unit
        at tests.Suite$$anonfun$test$1::apply$mcZ$sp_bool
        at tests.Suite$$anonfun$run$1::apply_class.tests.Test_unit
        at tests.Suite$$anonfun$run$1::apply_class.java.lang.Object_class.java.lang.Object
        at tests.Suite::run_bool
        at tests.Main$::main_class.ssnr.ObjectArray_unit
        at main
        at __libc_start_main
        at _start
        at 
```

with `ExceptionSuite.run()` in `Main`

Comment (densh) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: Can you print your clang invocation? It seemed like some calls were inlined. We use "-O0" by default in tests so this should not normally happen.

Comment (MasseGuillaume) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]:
```
> tests/run
[info] running
[info] /nix/store/p4zpakqa8rdhaz7cg6mjgib3688fy047-clang-wrapper-3.9.1/bin/clang++
[info]  -o
[info]  /home/gui/center/native/scala-native/unit-tests/target/scala-2.11/tests-out
[info]  -lrt
[info]  -lunwind
[info]  -lunwind-x86_64
[info]  -lgc
[info]  -target
[info]  x86_64-unknown-linux-gnu
[info]  -I/usr/local/include
[info]  -L/usr/local/lib
[info]  -O0
[info]  /home/gui/center/native/scala-native/unit-tests/target/scala-2.11/out.ll
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/gc.o
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/time.o
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/unwind.o
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/wrap.o
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/posix.o
[info]  /home/gui/.scalanative/nativelib-0.1.0-SNAPSHOT/eh.o
```

Comment (densh) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: This looks like a legitimate tests failure, we should not change the test here.

Comment (MasseGuillaume) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: @densh any hint on what went wrong here ?

Comment (MasseGuillaume) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: result:
```
at java.lang.ExceptionSuite$$anonfun$1::apply$mcV$sp_unit
at tests.Suite$$anonfun$test$1::apply$mcZ$sp_bool
at tests.Suite$$anonfun$run$1::apply_class.java.lang.Object_class.java.lang.Object
at scala.collection.mutable.UnrolledBuffer::foreach_trait.scala.Function1_unit
at tests.Suite::run_bool
at tests.Main$$anonfun$main$1::apply_class.java.lang.Object_class.java.lang.Object
at scala.collection.immutable.List::foreach_trait.scala.Function1_unit
at tests.Main$::main_class.ssnr.ObjectArray_unit
at main
at __libc_start_main
at _start
at 
```

expected
```
at java.lang.Throwable::fillInStackTrace_class.java.lang.Throwable
at java.lang.Throwable::init_class.java.lang.String_class.java.lang.Throwable
at java.lang.Exception::init_class.java.lang.String_class.java.lang.Throwable
at java.lang.Exception::init
at java.lang.ExceptionSuite$$anonfun$1::apply$mcV$sp_unit
at tests.Suite$$anonfun$test$1::apply$mcZ$sp_bool
at tests.Suite$$anonfun$run$1::apply_class.tests.Test_unit
at tests.Suite$$anonfun$run$1::apply_class.java.lang.Object_class.java.lang.Object
at scala.collection.mutable.UnrolledBuffer$Unrolled::foreach_trait.scala.Function1_unit
at scala.collection.mutable.UnrolledBuffer::foreach_trait.scala.Function1_unit
at tests.Suite::run_bool
at tests.Main$$anonfun$main$1::apply_class.tests.Suite_unit
at tests.Main$$anonfun$main$1::apply_class.java.lang.Object_class.java.lang.Object
at scala.collection.immutable.List::foreach_trait.scala.Function1_unit
at tests.Main$::main_class.ssnr.ObjectArray_unit
at main
```

 diff result expected
```
0a1,4
> at java.lang.Throwable::fillInStackTrace_class.java.lang.Throwable
> at java.lang.Throwable::init_class.java.lang.String_class.java.lang.Throwable
> at java.lang.Exception::init_class.java.lang.String_class.java.lang.Throwable
> at java.lang.Exception::init
2a7
> at tests.Suite$$anonfun$run$1::apply_class.tests.Test_unit
3a9
> at scala.collection.mutable.UnrolledBuffer$Unrolled::foreach_trait.scala.Function1_unit
5a12
> at tests.Main$$anonfun$main$1::apply_class.tests.Suite_unit
10,12d16
< at __libc_start_main
< at _start
< at 
```

Comment (densh) [inline unit-tests/src/main/scala/java/lang/ExceptionSuite.scala]: I don't exactly know the cause yet, lets file and discuss the issue separately.

**Comment by densh** [comment]
Ping @MasseGuillaume.

**Comment by densh** [comment]
LGTM!

**Comment by MasseGuillaume** [comment]
@densh the docker build fails at some point. I'm taking a look at it.

**Pull request merged**